pkg/controller/certificates/sync.go

```diff
@@ -164,13 +174,18 @@ func (c *Controller) Sync(ctx context.Context, crt *v1alpha1.Certificate) (reque
 	expectedCN := pki.CommonNameForCertificate(crtCopy)
 	expectedDNSNames := pki.DNSNamesForCertificate(crtCopy)
 
+	// as there is an existing certificate, or we may create one below, we will
+	// run scheduleRenewal to schedule a renewal if required at the end of
+	// execution.
+	defer c.scheduleRenewal(crtCopy)
```

Comment:
Hm. The fact that you've added this back here I think has just highlighted a bug in #788. This call only exists in the `issue` function in the current HEAD of master, meaning renewals will *not* be scheduled after cert-manager has restarted (or for pre-existing certificates).

I'll either put a patch in to resolve that now (which will cause a minor conflict), or maybe we can just get this PR merged to resolve the issue.

I'll also be building unit tests out for this soon, as this sort of thing should *definitely* be picked up by automation.

Reply:
BTW this section is the one that trips me up during rebasing, I'm not sure whats supposed to be happening and it seems to change a lot, can you verify that I got it right?

Reply:
So I recently made a couple of changes here to make this area of code easier to read.

For the time being, remove this altogether (i.e. leave as-is).

This is a separate bug, and it's one that can be fixed independently. I will create a PR shortly that fixes up this area of the code properly 👍

Reply:
So I should change it back or leave it as it is in the current commit?